crates/router/src/connector/checkout/transformers.rs

```diff
@@ -1,30 +1,150 @@
+                        wallet_data.get_wallet_token_as_json()?;
+                    Ok(Self {
+                        token_request_type: CheckoutTokenRequestType::Googlepay,
+                        token_data: CheckoutTokenData::Googlepay(json_wallet_data),
+                    })
+                }
+                api_models::payments::WalletData::ApplePay(_data) => {
+                    let json_wallet_data: CheckoutApplePayData =
+                        wallet_data.get_wallet_token_as_json()?;
+                    Ok(Self {
+                        token_request_type: CheckoutTokenRequestType::Applepay,
+                        token_data: CheckoutTokenData::Applepay(json_wallet_data),
+                    })
+                }
+                _ => Err(errors::ConnectorError::NotImplemented(
+                    "Unknown Wallet in Payment Method for tokenization".to_string(),
+                ))
+                .into_report(),
+            },
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "Unknown payment method for tokenization".to_string(),
+            ))
+            .into_report(),
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Deserialize)]
+pub struct CheckoutTokenResponse {
+    token: String,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, CheckoutTokenResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, CheckoutTokenResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: Ok(types::PaymentsResponseData::TokenizationResponse {
+                token: item.response.token,
+            }),
+            ..item.data
+        })
+    }
+}
+
 #[derive(Debug, Serialize)]
 pub struct CardSource {
     #[serde(rename = "type")]
-    pub source_type: Option<String>,
-    pub number: Option<pii::Secret<String, pii::CardNumber>>,
-    pub expiry_month: Option<pii::Secret<String>>,
-    pub expiry_year: Option<pii::Secret<String>>,
+    pub source_type: CheckoutSourceTypes,
+    pub number: pii::Secret<String, pii::CardNumber>,
+    pub expiry_month: pii::Secret<String>,
+    pub expiry_year: pii::Secret<String>,
+    pub cvv: pii::Secret<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct WalletSource {
+    #[serde(rename = "type")]
+    pub source_type: CheckoutSourceTypes,
+    pub token: String,
 }
 
 #[derive(Debug, Serialize)]
 #[serde(untagged)]
-pub enum Source {
+pub enum PaymentSource {
     Card(CardSource),
+    Wallets(WalletSource),
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CheckoutSourceTypes {
+    Card,
+    Token,
 }
 
 pub struct CheckoutAuthType {
     pub(super) api_key: String,
     pub(super) processing_channel_id: String,
+    pub(super) api_secret: String,
```

Comment:
update this change in public view sheet, we need to inform dashboard team about this change and get it changed in dashboard too. they won't take these changes right away so inform Anand about it

Reply:
Done the same.